src/components/join-list/index.js

```diff
@@ -0,0 +1,46 @@
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import Button from '../button';
+
+export const JoinList = () => {
+  const history = useHistory();
+  const { setValue } = useLocalStorage();
+  const [listToken, setListToken] = useState('');
+
+  const retrieveList = (e) => {
+    e.preventDefault();
+    setValue(listToken);
+    history.push('/list');
```

Comment:
I believe that before you redirect to `/list` could be interesting to check if this token is valid or not. Because the app allows you to change between pages and do some operations like `list` & `add` products but with some errors.

Also, I think we need to add some extra controls in `list` and `add` item pages. 

I have some ideas or possible solutions I'm going to write them in my final comment about my first review.

Reply:
**Browser console** 👀 

https://user-images.githubusercontent.com/8689897/138389276-8a573971-ff86-410c-a177-19311925baf8.mp4